### PR TITLE
CLI Syntax Guidelines

### PR DESCRIPTION
**Pull request by KathleenDollard**
Guidelines for syntax for the .NET Core CLI starting in 2.1.300. 

This describes a change in thinking that will affect certain new syntax. 
 
There are no breaking changes in previously released syntax. 

There is a breaking change between .NET Core SDK 2.1.300 Preview1 and Preview2

### CODE REVIEW

**Comment by omajid** [comment]
This is fantastic!

File: proposed/cli-syntax-guideline.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+```
+
+The core context for the .NET Core CLI is the project. Like many other CLIs, this concept is a little fluffy and you will probably comfortably think about it as _the main thing I am working on right now_. This will generally be a project, but it might in some cases be a solution or a directory. We think of a solution as a super-project and in some cases a project is represented by its directory. The .NET Core CLI will adopt the Docker style of allowing the core context to be explicit or implicit because it makes help much easier. `dotnet --help` may have 30 entries while `dotnet project --help` may have 15. For example, the following commands continue to work as before:
+
+``` terminal
+$ build
+$ new
+$ publish
+$ restore
+$ run
```
Comment (omajid) [inline proposed/cli-syntax-guideline.md]: This should be `dotnet run` as opposed to `run`, right?

File: proposed/cli-syntax-guideline.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+```
+
+The following commands will work in a future version of the .NET Core CLI:
+
+``` terminal
+$ project build
+$ project new
+$ project publish
+$ project restore
+$ project run
```
Comment (omajid) [inline proposed/cli-syntax-guideline.md]: This should be `dotnet project run`, right?

File: proposed/cli-syntax-guideline.md (hunk 1 above)
Comment (KathleenDollard) [reply proposed/cli-syntax-guideline.md]: Yes, thank you for catching it!!!

**Pull request merged**